src/Cmfd.cpp

```diff
@@ -1750,22 +1774,278 @@ void Cmfd::setPolarQuadrature(PolarQuad* polar_quad) {
+      if (iter->first == 0)
+        ratio += (1.0 - iter->second/total_distance) *
+          getFluxRatio(cmfd_cell - _num_x - 1, moc_group);
+
+      /* BOTTOM SIDE */
+      else if (iter->first == 1)
+        ratio += (1.0 - iter->second/total_distance) *
+          getFluxRatio(cmfd_cell - _num_x, moc_group);
+
+      /* LOWER RIGHT CORNER */
+      else if (iter->first == 2)
+        ratio += (1.0 - iter->second/total_distance) *
+          getFluxRatio(cmfd_cell - _num_x + 1, moc_group);
+
+      /* LEFT SIDE */
+      else if (iter->first == 3)
+        ratio += (1.0 - iter->second/total_distance) *
+          getFluxRatio(cmfd_cell - 1, moc_group);
+
+      /* RIGHT SIDE */
+      else if (iter->first == 5)
+        ratio += (1.0 - iter->second/total_distance) *
+          getFluxRatio(cmfd_cell + 1, moc_group);
+
+      /* UPPER LEFT CORNER */
+      else if (iter->first == 6)
+        ratio += (1.0 - iter->second/total_distance) *
+          getFluxRatio(cmfd_cell + _num_x - 1, moc_group);
+
+      /* TOP SIDE */
+      else if (iter->first == 7)
+        ratio += (1.0 - iter->second/total_distance) *
+          getFluxRatio(cmfd_cell + _num_x, moc_group);
+
+      /* TOP RIGHT CORNER */
+      else if (iter->first == 8)
+        ratio += (1.0 - iter->second/total_distance) *
+          getFluxRatio(cmfd_cell + _num_x + 1, moc_group);
+    }
+
+    /* INTERNAL */
+    if (_k_nearest_stencils[fsr].size() == 1)
+      ratio += getFluxRatio(cmfd_cell, moc_group);
+    else{
+      ratio += (1.0 - _k_nearest_stencils[fsr][0].second/total_distance) *
+        getFluxRatio(cmfd_cell, moc_group);
+      ratio /= (_k_nearest_stencils[fsr].size() - 1);
+    }
+  }
+  else
+    ratio = getFluxRatio(cmfd_cell, moc_group);
+
   return ratio;
 }
 
 
 /**
+ * @brief Get the distances from an FSR centroid to a given cmfd cell.
+ * @detail This method takes in a FSR centroid, a cmfd cell, and a stencil index
+ *         to a cell located in the 9-point stencil encompassing the cmfd
+ *         cell an all its possible neighbors. The CMFD cell stencil is:
+ *
+ *                             6 7 8
+ *                             3 4 5
+ *                             0 1 2
+ *
+ *         where 4 is the given CMFD cell. If a CMFD edge or corner cells is
+ *         given and the stencil indexed cell lies outside the geometry, the
+ *         maximum allowable FP_PRECISION value is returned.
+ * @param centroid The numerical centroid an FSR in the cell.
+ * @param cell The cmfd cell containing the FSR.
+ * @param stencil_index The index of the cell in the stencil that we want to
+ *        get the distance from.
+ * @return the distance from the CMFD cell centroid to the FSR centroid.
+ */
+FP_PRECISION Cmfd::getDistanceToCentroid(Point* centroid, int cell,
+                                         int stencil_index){
+
+  int x = cell % _num_x;
+  int y = cell / _num_x;
+  FP_PRECISION dist_x, dist_y;
+  bool found = false;
```

Comment:
This routine seems fine to me, just curious if it would be impossible (or just not worth it) to use one of the ``Point::distance(double x, double y)`` or ``Point::distianceToPoint(Point* point)`` routines here?